go/pkg/middleware/proxy_test.go

```diff
@@ -202,6 +202,35 @@ func TestReverseProxyAuthenticator_badtoken(t *testing.T) {
 	assert.Equal(t, "bad authentication token", getStrippedBody(resp))
 }
 
+// func TestReverseProxy_CortexPushRewrite429(t *testing.T) {
```

Comment:
todo: test this rewrite. wasn't obvious where to test this really -- add a new _test file to `cmd/cortex/..`?